Ai fleet merger cleanup

Dilvish-fo opened the pull request:
A bit of cleanup related to AI fleet mergers and splits.  Among other things, this appears to resolve #1794

Dilvish-fo committed:
further restrict AI fleet mergers, to reduce fleet object churn

Morlic-fo reviewed (commented):
Some style nitpicking otherwise looks ok.

Dilvish-fo committed:
Adjust AI fleet merge handling
 - replace a monster check with a speed check
 - make AI deletion of stale fleet info more thorough
